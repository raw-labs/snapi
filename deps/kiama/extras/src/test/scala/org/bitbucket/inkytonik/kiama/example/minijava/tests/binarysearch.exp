20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
99999
0
0
1
1
1
1
0
0
999
