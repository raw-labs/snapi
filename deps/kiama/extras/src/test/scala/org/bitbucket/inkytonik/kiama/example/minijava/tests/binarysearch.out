.source src/test/scala/org/bitbucket/inkytonik/kiama/example/minijava/tests/binarysearch.mj
.class public BinarySearch
.super java/lang/Object

.method <init>()V
.limit stack 1
.limit locals 1
    aload_0
    invokespecial java/lang/Object/<init>()V
    return
.end method

.method public static main([Ljava/lang/String;)V
.limit stack 3
.limit locals 2
    getstatic java/lang/System/out Ljava/io/PrintStream;
    new BS
    dup
    invokespecial BS/<init>()V
    bipush 20
    invokevirtual BS/Start(I)I
    invokevirtual java/io/PrintStream/println(I)V
    return
.end method

.source src/test/scala/org/bitbucket/inkytonik/kiama/example/minijava/tests/binarysearch.mj
.class public BS
.super java/lang/Object
.field public number [I
.field public size I

.method <init>()V
.limit stack 1
.limit locals 1
    aload_0
    invokespecial java/lang/Object/<init>()V
    return
.end method

.method public Start(I)I
.limit stack 2
.limit locals 4
    aload 0
    iload 1
    invokevirtual BS/Init(I)I
    istore 2
    aload 0
    invokevirtual BS/Print()I
    istore 3
    aload 0
    bipush 8
    invokevirtual BS/Search(I)Z
    ifeq L1
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_1
    invokevirtual java/io/PrintStream/println(I)V
    goto L2
L1:
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_0
    invokevirtual java/io/PrintStream/println(I)V
L2:
    aload 0
    bipush 19
    invokevirtual BS/Search(I)Z
    ifeq L3
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_1
    invokevirtual java/io/PrintStream/println(I)V
    goto L4
L3:
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_0
    invokevirtual java/io/PrintStream/println(I)V
L4:
    aload 0
    bipush 20
    invokevirtual BS/Search(I)Z
    ifeq L5
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_1
    invokevirtual java/io/PrintStream/println(I)V
    goto L6
L5:
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_0
    invokevirtual java/io/PrintStream/println(I)V
L6:
    aload 0
    bipush 21
    invokevirtual BS/Search(I)Z
    ifeq L7
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_1
    invokevirtual java/io/PrintStream/println(I)V
    goto L8
L7:
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_0
    invokevirtual java/io/PrintStream/println(I)V
L8:
    aload 0
    bipush 37
    invokevirtual BS/Search(I)Z
    ifeq L9
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_1
    invokevirtual java/io/PrintStream/println(I)V
    goto L10
L9:
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_0
    invokevirtual java/io/PrintStream/println(I)V
L10:
    aload 0
    bipush 38
    invokevirtual BS/Search(I)Z
    ifeq L11
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_1
    invokevirtual java/io/PrintStream/println(I)V
    goto L12
L11:
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_0
    invokevirtual java/io/PrintStream/println(I)V
L12:
    aload 0
    bipush 39
    invokevirtual BS/Search(I)Z
    ifeq L13
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_1
    invokevirtual java/io/PrintStream/println(I)V
    goto L14
L13:
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_0
    invokevirtual java/io/PrintStream/println(I)V
L14:
    aload 0
    bipush 50
    invokevirtual BS/Search(I)Z
    ifeq L15
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_1
    invokevirtual java/io/PrintStream/println(I)V
    goto L16
L15:
    getstatic java/lang/System/out Ljava/io/PrintStream;
    iconst_0
    invokevirtual java/io/PrintStream/println(I)V
L16:
    ldc 999
    ireturn
.end method

.method public Search(I)Z
.limit stack 8
.limit locals 9
    iconst_0
    istore 2
    iconst_0
    istore 3
    aload 0
    getfield BS/number [I
    arraylength
    istore 4
    iload 4
    iconst_1
    isub
    istore 4
    iconst_0
    istore 5
    iconst_1
    istore 6
L1:
    iload 6
    ifeq L2
    iload 5
    iload 4
    iadd
    istore 7
    aload 0
    iload 7
    invokevirtual BS/Div(I)I
    istore 7
    aload 0
    getfield BS/number [I
    iload 7
    iaload
    istore 2
    iload 1
    iload 2
    if_icmpge L5
    iconst_1
    goto L6
L5:
    iconst_0
L6:
    ifeq L3
    iload 7
    iconst_1
    isub
    istore 4
    goto L4
L3:
    iload 7
    iconst_1
    iadd
    istore 5
L4:
    aload 0
    iload 2
    iload 1
    invokevirtual BS/Compare(II)Z
    ifeq L7
    iconst_0
    istore 6
    goto L8
L7:
    iconst_1
    istore 6
L8:
    iload 4
    iload 5
    if_icmpge L11
    iconst_1
    goto L12
L11:
    iconst_0
L12:
    ifeq L9
    iconst_0
    istore 6
    goto L10
L9:
    iconst_0
    istore 8
L10:
    goto L1
L2:
    aload 0
    iload 2
    iload 1
    invokevirtual BS/Compare(II)Z
    ifeq L13
    iconst_1
    istore 3
    goto L14
L13:
    iconst_0
    istore 3
L14:
    iload 3
    ireturn
.end method

.method public Div(I)I
.limit stack 3
.limit locals 5
    iconst_0
    istore 2
    iconst_0
    istore 3
    iload 1
    iconst_1
    isub
    istore 4
L1:
    iload 3
    iload 4
    if_icmpge L3
    iconst_1
    goto L4
L3:
    iconst_0
L4:
    ifeq L2
    iload 2
    iconst_1
    iadd
    istore 2
    iload 3
    iconst_2
    iadd
    istore 3
    goto L1
L2:
    iload 2
    ireturn
.end method

.method public Compare(II)Z
.limit stack 4
.limit locals 5
    iconst_0
    istore 3
    iload 2
    iconst_1
    iadd
    istore 4
    iload 1
    iload 2
    if_icmpge L3
    iconst_1
    goto L4
L3:
    iconst_0
L4:
    ifeq L1
    iconst_0
    istore 3
    goto L2
L1:
    iload 1
    iload 4
    if_icmpge L9
    iconst_1
    goto L10
L9:
    iconst_0
L10:
    ifeq L7
    iconst_0
    goto L8
L7:
    iconst_1
L8:
    ifeq L5
    iconst_0
    istore 3
    goto L6
L5:
    iconst_1
    istore 3
L6:
L2:
    iload 3
    ireturn
.end method

.method public Print()I
.limit stack 6
.limit locals 2
    iconst_1
    istore 1
L1:
    iload 1
    aload 0
    getfield BS/size I
    if_icmpge L3
    iconst_1
    goto L4
L3:
    iconst_0
L4:
    ifeq L2
    getstatic java/lang/System/out Ljava/io/PrintStream;
    aload 0
    getfield BS/number [I
    iload 1
    iaload
    invokevirtual java/io/PrintStream/println(I)V
    iload 1
    iconst_1
    iadd
    istore 1
    goto L1
L2:
    getstatic java/lang/System/out Ljava/io/PrintStream;
    ldc 99999
    invokevirtual java/io/PrintStream/println(I)V
    iconst_0
    ireturn
.end method

.method public Init(I)I
.limit stack 8
.limit locals 6
    aload 0
    iload 1
    putfield BS/size I
    aload 0
    iload 1
    newarray int
    putfield BS/number [I
    iconst_1
    istore 2
    aload 0
    getfield BS/size I
    iconst_1
    iadd
    istore 3
L1:
    iload 2
    aload 0
    getfield BS/size I
    if_icmpge L3
    iconst_1
    goto L4
L3:
    iconst_0
L4:
    ifeq L2
    iconst_2
    iload 2
    imul
    istore 4
    iload 3
    iconst_3
    isub
    istore 5
    aload 0
    getfield BS/number [I
    iload 2
    iload 4
    iload 5
    iadd
    iastore
    iload 2
    iconst_1
    iadd
    istore 2
    iload 3
    iconst_1
    isub
    istore 3
    goto L1
L2:
    iconst_0
    ireturn
.end method

